src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -711,24 +713,30 @@ private async ValueTask<bool> HandleSetColor(ClientPlayer sender, ColorType colo
             }
             else
             {
-                if (!RequestedColorId.Any())
+                if (RequestedColorId.Any())
                 {
-                    _logger.LogWarning($"Client sent {nameof(RpcCalls.SetColor)} for a player that didn't request it");
-                    return false;
-                }
+                    var expected = RequestedColorId.Dequeue();
 
-                var expected = RequestedColorId.Dequeue();
+                    while (Game.Players.Any(x => x.Character != null && x.Character != this && x.Character.PlayerInfo.CurrentOutfit.Color == expected))
```

Comment:
Technically this can infinitely loop when all ColorTypes are picked. I guess the loop is trying to assign a random new color if the player sends one that is already in use? What if it randomly assigns one that another player in the next packet wanted?

(I see that it's not a change of this PR but still)